src/bosh_azure_cpi/lib/cloud/azure/vm_manager.rb

```diff
@@ -4,33 +4,66 @@ module Bosh::AzureCloud
   class VMManager
     include Helpers
 
-    def initialize(azure_config, registry_endpoint, disk_manager, disk_manager2, azure_client2, storage_account_manager)
+    def initialize(azure_config, registry_endpoint, disk_manager, disk_manager2, azure_client2, storage_account_manager, stemcell_manager, stemcell_manager2, light_stemcell_manager)
       @azure_config = azure_config
       @registry_endpoint = registry_endpoint
       @disk_manager = disk_manager
       @disk_manager2 = disk_manager2
       @azure_client2 = azure_client2
       @storage_account_manager = storage_account_manager
-      @use_managed_disks = @azure_config['use_managed_disks']
+      @stemcell_manager = stemcell_manager
+      @stemcell_manager2 = stemcell_manager2
+      @light_stemcell_manager = light_stemcell_manager
+      @use_managed_disks = azure_config['use_managed_disks']
       @logger = Bosh::Clouds::Config.logger
     end
 
-    def create(instance_id, location, stemcell_info, vm_properties, network_configurator, env)
+    def create(instance_id, location, stemcell_id, vm_properties, network_configurator, env)
       # network_configurator contains service principal in azure_config so we must not log it.
-      @logger.info("create(#{instance_id}, #{location}, #{stemcell_info.inspect}, #{vm_properties}, ..., ...)")
```

Comment:
stemcell_info.inspect will be lost in the log.
maybe we should log it down in other place?

Reply:
log it in get_stemcell_info